specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/compute.json

```diff
@@ -12067,22 +12078,44 @@
         "action"
       ]
     },
-    "PatchSettings": {
+    "WindowsPatchSettings": {
       "properties": {
         "patchMode": {
           "type": "string",
-          "description": "Specifies the mode of in-guest patching to IaaS virtual machine.<br /><br /> Possible values are:<br /><br /> **Manual** - You  control the application of patches to a virtual machine. You do this by applying patches manually inside the VM. In this mode, automatic updates are disabled; the property WindowsConfiguration.enableAutomaticUpdates must be false<br /><br /> **AutomaticByOS** - The virtual machine will automatically be updated by the OS. The property WindowsConfiguration.enableAutomaticUpdates must be true. <br /><br /> ** AutomaticByPlatform** - the virtual machine will automatically updated by the platform. The properties provisionVMAgent and WindowsConfiguration.enableAutomaticUpdates must be true ",
+          "description": "Specifies the mode of VM Guest Patching to IaaS virtual machine.<br /><br /> Possible values are:<br /><br /> **Manual** - You  control the application of patches to a virtual machine. You do this by applying patches manually inside the VM. In this mode, automatic updates are disabled; the property WindowsConfiguration.enableAutomaticUpdates must be false<br /><br /> **AutomaticByOS** - The virtual machine will automatically be updated by the OS. The property WindowsConfiguration.enableAutomaticUpdates must be true. <br /><br /> ** AutomaticByPlatform** - the virtual machine will automatically updated by the platform. The properties provisionVMAgent and WindowsConfiguration.enableAutomaticUpdates must be true ",
           "enum": [
             "Manual",
             "AutomaticByOS",
             "AutomaticByPlatform"
           ],
           "x-ms-enum": {
-            "name": "InGuestPatchMode",
+            "name": "WindowsVMGuestPatchMode",
             "modelAsString": true
           }
+        },
+        "enableHotpatching": {
+          "type": "boolean",
+          "description": "Enables customers to patch their Azure VMs without requiring a reboot. For enableHotpatching, the 'provisionVMAgent' must be set to true and 'patchMode' must be set to 'AutomaticByPlatform'."
```

Comment:
@najams - note on description here. Hotpatching still requires provisionVMagent in the code. But it could be made to match regular auto-patching on AllowExtensionOperations (as in the comment for that). Do we want to proactively match the description here.

Reply:
Again, from the discussion with Najam, it is okay to keep it as is to keep it simple for user i.e. always enable provisionVMAgent.